### DISCUSSION TITLE
Hazel-like keyword bubbles

### DISCUSSION THREAD

**Discussion by Felienne** [hedyorg/hedy#4674] [Large Projects & Plans]
Hazel is a functional programming language with an AMAZING UX, you can play with it here: https://hazel.org/build/dev/

There are a lot of things we will not be able to do in Hedy because it is not a functional typed language, but one idea could be useful.... the nice little bubbles that they show, which are keywords that still need to be placed, like this:

<img width="581" alt="image" src="https://github.com/hedyorg/hedy/assets/1003685/58ca076e-ccc4-4c50-af2a-8d11e93a010d">

We very often see kids forgetting things like "times" after repeat, and such a bubble is a very elegant way to show what they still need to do, which is a bit like an error message, but pre-emptively and not "angry"

**Comment by jpelay** [comment]
Wow! What a cool interface!! We can do something like that using CodeMirror, I believe. We can build autocomplete features for some of the keywords: https://codemirror.net/examples/autocompletion/
